src/EditorFeatures/TestUtilities/Formatting/CoreFormatterTestsBase.cs

```diff
@@ -40,20 +36,8 @@ public abstract class CoreFormatterTestsBase
             var snapshot = subjectDocument.TextBuffer.CurrentSnapshot;
             var indentationLineFromBuffer = snapshot.GetLineFromPosition(point);
 
-            var commandHandler = CreateSmartTokenFormatterCommandHandler(textUndoHistory.Object, editorOperationsFactory.Object);
-            commandHandler.ExecuteCommandWorker(new ReturnKeyCommandArgs(textView, subjectDocument.TextBuffer), CancellationToken.None);
-            var newSnapshot = subjectDocument.TextBuffer.CurrentSnapshot;
-
-            int? actualIndentation;
-            if (newSnapshot.Version.VersionNumber > snapshot.Version.VersionNumber)
-            {
-                actualIndentation = newSnapshot.GetLineFromLineNumber(indentationLineFromBuffer.LineNumber).GetFirstNonWhitespaceOffset();
-            }
-            else
-            {
-                var provider = new SmartIndent(textView);
-                actualIndentation = provider.GetDesiredIndentation(indentationLineFromBuffer);
-            }
+            var provider = new SmartIndent(textView);
+            var actualIndentation = provider.GetDesiredIndentation(indentationLineFromBuffer);
```

Comment:
these tests were super strange.  They would try to replicate that SmartIndent and SmartTokenFormatCommandHandler were living side-by-side and handlign the same responsibility.

Reply:
so SmartIndent never changes buffer but SmartTokenFormatter does.

I am not sure how SmartIndent can do what SmartTokenFormatter does...

I can't remember all different cases but, something like

void Method()  | <= enter here { 

regular smart indenter do something like
void Method()
| <= caret { 

which "{" is at wrong place, 

so, smart token formatter override default smart indenter behavior (again, like I said multiple time, I dont know who made this decision or since when we had this behavior, I just know this should be at least since orcas or whidbey)

it basically edit buffer so that "{" goes to like place like

void Method()
|{ <= here

this test is making sure that edit happened so that things are at right location.

Reply:
So, trying this all out, it doesn't seem like any special behavior handling is necessary.  The behavior you describe is what we get:

![Enter](https://user-images.githubusercontent.com/4564579/58516794-11612980-815e-11e9-8d85-c4d7c9e67a5e.gif)

This has been my general frustration with the complexity here (and why i've been making all these PRs).  Basically, i don't think we need any of this split.  We just need a way to say "give me the expected indentation here".  That, under the covers, can defer to a few different impls with differnet smarts.  but no one should need to know the internal details, just that they got an indentation result.

Reply:
interesting. so looked.

it looks like it works now since you always return indentation. so undo behavior is changed
![format_undo1](https://user-images.githubusercontent.com/1333179/59385778-f673e500-8d19-11e9-97ee-0dc59b4cc3ee.gif)
to
![format_undo2](https://user-images.githubusercontent.com/1333179/59385803-02f83d80-8d1a-11e9-96c8-dad9c97ed50a.gif)

for whatever legacy reason, we always preserved 2 undo for auto formatting. so, that is now gone.

no idea whether we care about it or not, but it certainly change of behavior. not sure whether we need to bring in that don't ever break muscle memory thing here or not.

also, existing code is considered to work when smart indentation and auto formatting is all turned off for whatever reason since option is provided. and preserve pre Roslyn behavior. and that is changed now.

![format_noautoformat1](https://user-images.githubusercontent.com/1333179/59392254-d2210400-8d2b-11e9-88a5-ce4308af43fc.gif)
vs
![format_noautoformat2](https://user-images.githubusercontent.com/1333179/59392257-d816e500-8d2b-11e9-98ea-8b9ca064f43e.gif)

who knows what some random test in a world this behavior change might break.

anyway, I think I wrote it somewhere but ISmartIndenter is editor service that we implemented specifically for editor. and we added IIndentationService that is a plain service. whether 2 needs to be same or not. 

I guess that's up to the team. tagging @jinujoseph

Reply:
alright, can't remember all different combinations of options between indentation, auto formatting, brace completion whether each affects each other's behaviors. currently, I believe behavior try to preserve pre Roslyn behavior as much as possible in editor.

whether we want or need to preserve all those behavior? hard to know. we could try and see how much feedback we get.

Reply:
Good cases. Note: these all appear to be the expected behavior for the IIndentService.